ras_party/support/util.py

```diff
@@ -64,7 +68,20 @@ def obfuscate_email(email):
     """Takes an email address and returns an obfuscated version of it.
     For example: test@example.com would turn into t**t@e*********m
     """
-    m = email.split('@')
-    prefix = f'{m[0][0]}{"*"*(len(m[0])-2)}{m[0][-1]}'
-    domain = f'{m[1][0]}{"*"*(len(m[1])-2)}{m[1][-1]}'
-    return f'{prefix}@{domain}'
+    try:
+        m = email.split('@')
+        # If the prefix is 1 character, then we can't obfuscate it
+        if len(m[0]) > 1:
+            prefix = f'{m[0][0]}{"*"*(len(m[0])-2)}{m[0][-1]}'
+        else:
+            prefix = m[0]
+        # If the domain is missing or 1 character, then we can't obfuscate it
+        if len(m) > 1 and len(m[1]) > 1:
+            domain = f'{m[1][0]}{"*"*(len(m[1])-2)}{m[1][-1]}'
+            return f'{prefix}@{domain}'
+        else:
+            return f'{prefix}'
+    except AssertionError:
```

Comment:
How does the assertionError get raised?

Reply:
That was also raised by test_responded_controller.py (see above), so that may also be overkill.

Reply:
I've replaced the try-except with a simpler set of ifs.